src/GraphQL/loaders/parseClassMutations.js

```diff
@@ -138,19 +173,34 @@ const load = function(
       args: {
         objectId: defaultGraphQLTypes.OBJECT_ID_ATT,
       },
-      type: new GraphQLNonNull(GraphQLBoolean),
-      async resolve(_source, args, context) {
+      type: new GraphQLNonNull(classGraphQLOutputType),
+      async resolve(_source, args, context, mutationInfo) {
         try {
           const { objectId } = args;
           const { config, auth, info } = context;
-
-          return await objectsMutations.deleteObject(
+          const selectedFields = getFieldNames(mutationInfo);
+          const { keys, include } = parseClassTypes.extractKeysAndInclude(
+            selectedFields
+          );
+          const object = await objectsQueries.getObject(
```

Comment:
I'd only do the query if the request is asking more than objecId

Reply:
Got it